ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManagerMetrics.java

```diff
@@ -190,15 +196,104 @@ void initializePartitionMetrics() {
           }
         }
         if (downReplicaCount > replicaCount / 2) {
-            logger.info("There are more than half of the replicas are down for partition {}, the down replicas are {}",
-                partition.toPathString(), downReplicas);
+          logger.info("There are more than half of the replicas are down for partition {}, the down replicas are {}",
+              partition.toPathString(), downReplicas);
           return 1L;
         }
       }
       return 0L;
     };
     registry.gauge(MetricRegistry.name(HelixClusterManager.class, "isMajorityReplicasDownForAnyPartition"),
         () -> isMajorityReplicasDownForAnyPartition);
+
+    Gauge<Integer> getLocalPartitionCountWithOneReplicaDown = () -> {
+      String dcName = clusterMapCallback.getDatacenterName();
+
+      for (PartitionId partition : clusterMapCallback.getPartitions()) {
+        // Get total replica count
+        int totalLocalReplicaCount = (int) partition.getReplicaIds()
+            .stream()
+            .filter(replicaId -> replicaId.getDataNodeId().getDatacenterName().equals(dcName))
+            .count();
+
+        // Get local leader/standby/bootstrap replica counts of this partition
+        int localLeaderStandByReplicaCount = 0;
+        for (List<? extends ReplicaId> replicaIds : partition.getReplicaIdsByStates(
+            EnumSet.of(ReplicaState.LEADER, ReplicaState.STANDBY), dcName).values()) {
+          localLeaderStandByReplicaCount += replicaIds.size();
+        }
+        int localBootstrapReplicaCount = partition.getReplicaIdsByState(ReplicaState.BOOTSTRAP, dcName).size();
+        int localInactiveReplicaCount = partition.getReplicaIdsByState(ReplicaState.INACTIVE, dcName).size();
+
+        // Calculate the number of replicas that are down. Bootstrap (upcoming) and Inactive (deactivating) replicas are considered as good.
+        int downReplicaCount = totalLocalReplicaCount - localLeaderStandByReplicaCount - localBootstrapReplicaCount
```

Comment:
Shouldn't we count Inactive replicas as part of our  down states count so we have visibility on processes that gracefully shutdown the apps

Reply:
Sorry, didn't get you. In `partitionCountWithOneReplicaDown`, we ignore inactive replicas but include them in  `partitionCountWithTwoNonLeaderStandByReplica`. I am thinking that using both the metrics, we could get idea on the partitions which are unhealthy.